src/loader/fragment.ts

```diff
@@ -302,9 +302,7 @@ export class Part extends BaseSegment {
     super(baseurl);
     this.duration = partAttrs.decimalFloatingPoint('DURATION');
     this.gap = partAttrs.bool('GAP');
-    this.independent = partAttrs.INDEPENDENT
-      ? partAttrs.bool('INDEPENDENT')
-      : true;
+    this.independent = partAttrs.INDEPENDENT === 'YES';
```

Comment:
If it's important that the part object represents what is found in playlists, we can change `part.independent` to be optional

`public independent?: boolean;`

then only set when the attribute is present. 

```suggestion
    if (partAttrs.INDEPENDENT) {
      this.independent = partAttrs.bool('INDEPENDENT');
    }
```

Then we just need to change the logic in the stream controller to `part.independent
 !== false`:
https://github.com/video-dev/hls.js/blob/v1.0.0-rc.2/src/controller/base-stream-controller.ts#L782

Reply:
I disagree. `INDEPENDENT` only has one valid value: `YES`, which means that the part definitely includes a keyframe. _`NO` is not a valid value_!

Translating `YES` to `independent = true` makes sense, while any other value or omission must mean **maybe**, and is nicely represented by `independent = false`.

If a _completed_ segment contains no parts with ´independent: true`, you could try to do some recovery logic, but IMO the player should just reject the playlist for low-latency use, and play it as a non-low-latency stream.

Reply:
> `INDEPENDENT` only has one valid value: `YES`, which means that the part definitely includes a keyframe. `NO` is not a valid value!

Makes sense given the spec only mentions a value of `YES`.

```
INDEPENDENT

      The value is an enumerated-string whose value is YES if the
      Partial Segment contains an independent frame.  This attribute is
      OPTIONAL; however every Partial Segment containing an independent
      frame SHOULD carry it to increase the efficiency with which
      clients can join and switch Renditions.
```

These changes work fine with the sample Apple streams (which include the attribute). I'd like to get some input from other vendors before merging.

Reply:
If you really want to try to handle it when no part is flagged as independent, the fallback should probably be to handle it as if the first part is independent. That way a random access will be ensured to hit the keyframe that is somewhere inside the fragment.